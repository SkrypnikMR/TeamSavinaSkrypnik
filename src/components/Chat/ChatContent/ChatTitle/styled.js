import styled from 'styled-components';
import { bgColorDefault, bgColorDefaultFon, colorDefault } from '/src/components/UI/baseLayout';

export const StChatTitle = styled.div`
    display: flex;
    width: 100%;
    min-height: 50px;
    justify-content: space-between;
    align-items: center;
    color: ${({ color = colorDefault }) => color};
    font-size: 24px;
    font-family: 'Play', sans-serif;
    border-top: 1px solid black;
    border-bottom: 1px solid black;
    border-right: 1px solid black;
    padding: 5px 0;
    background-color: ${({ color = bgColorDefaultFon }) => color};
        span{
          color: ${({ color = bgColorDefault }) => color};
          margin-left: 20px;
        }
        p{
          cursor: pointer;
          margin-right: 20px;
          font-size: 16px;
        }
`;
