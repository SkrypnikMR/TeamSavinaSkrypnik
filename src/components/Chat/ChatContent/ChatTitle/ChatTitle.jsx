import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { StChatTitle } from './styled';

const ChatTitle = ({ currentRoomName, changeModalVisibility, getAllUsers }) => {
    const { t } = useTranslation();
    const handleOpenModalInviteUsers = () => {
        getAllUsers();
        changeModalVisibility({
            isOpen: true, data: {}, modalType: 'usersInChat',
        });
    };
    return (
        <StChatTitle>
            <span>
                {currentRoomName}
            </span>
            {currentRoomName && currentRoomName !== 'global'
                ? <p onClick={handleOpenModalInviteUsers}>{t('user_list')}</p>
                : null}
        </StChatTitle>
    );
};

ChatTitle.propTypes = {
    currentRoomName: PropTypes.string,
    changeModalVisibility: PropTypes.func,
    getAllUsers: PropTypes.func,
};

export default ChatTitle;
