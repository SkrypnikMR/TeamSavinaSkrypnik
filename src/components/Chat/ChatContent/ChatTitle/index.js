import { connect } from 'react-redux';
import ChatTitle from './ChatTitle.jsx';
import { currentRoomName } from '/src/Store/chat/selectors';
import { changeModalVisibility } from '/src/Store/modals/actions';
import { getAllUsers } from '/src/Store/chat/actions';

const mapStateToProps = state => ({
    currentRoomName: currentRoomName(state),
});

const mapDispatchToProps = dispatch => ({
    changeModalVisibility: payload => dispatch(changeModalVisibility(payload)),
    getAllUsers: () => dispatch(getAllUsers()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ChatTitle);
