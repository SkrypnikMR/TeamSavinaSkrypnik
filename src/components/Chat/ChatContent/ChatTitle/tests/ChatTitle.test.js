import React from 'react';
import { shallow } from 'enzyme';

import ChatTitle from '../ChatTitle';

describe('ChatTitle', () => {
  let props;
  const currentRoomName = 'global';
  beforeEach(() => {
    props = {
      currentRoomName,
    };
  });
  it('Should match snapshot', () => {
    const component = shallow(<ChatTitle {...props} />);
    expect(component.html).toMatchSnapshot();
  });
  it('Should render StChatTitle', () => {
    const component = shallow(<ChatTitle />);
    expect(component.find('styled__StChatTitle')).toHaveLength(1);
  });
  it('Should render p', () => {
    const component = shallow(<ChatTitle />);
    expect(component.find('p')).toHaveLength(1);
  });
  it('Should render span', () => {
    const component = shallow(<ChatTitle />);
    expect(component.find('span')).toHaveLength(1);
  });
});
