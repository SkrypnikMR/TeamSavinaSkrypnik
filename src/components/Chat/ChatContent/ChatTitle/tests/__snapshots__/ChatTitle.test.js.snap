// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatTitle Should match snapshot 1`] = `[Function]`;
